.github/workflows/pull-request-ci.yaml

```diff
@@ -171,15 +171,23 @@ jobs:
           echo "SSH_AGENT_PID=${SSH_AGENT_PID}" >> "${GITHUB_ENV}"
         timeout-minutes: 1
 
-      # set GKE secret. TBD: use vault instead
+      - name: Query vault for GKE secrets
+        id: vault
+        uses: hashicorp/vault-action@v2.0.1
+        with:
+          exportEnv: false
+          url: https://volt.cap.explore.suse.dev/
+          token: ${{ secrets.VAULT_TOKEN_CLOUD_PROVIDERS }}
+          secrets: |
+            secret/data/cloud-providers gke.GKE_CRED_JSON | GKE_CRED_JSON
+
+      # set GKE secret.
       - name: set GKE_CRED_JSON
         if: env.BACKEND == 'gke'
         run: |
           json_file="$(mktemp)"
-          echo "$GKE_CRED_JSON" > "${json_file}"
+          echo "${{ steps.vault.outputs.GKE_CRED_JSON }}" > "${json_file}"
```

Comment:
Okay, I actually tried to deploy this and this _definitely_ doesn't work.
The vault output contains the quote character (`"`), and because the `${{}}` is interpreted by GitHub, the shell command ends up being:
```bash
echo "{"blah": "foo", …}"
```
Of course, this means that the actual command run is (replacing the quoting):
```bash
echo '{blah: foo, …}'
```
That's no longer valid JSON, so it falls over in later steps.

Reply:
So, if I understand correct, I should remove the outer quotes ?
IOW

> echo ${{ steps.vault.outputs.GKE_CRED_JSON }} > "${json_file}"

On the other side, I see

> cat > "${json_file}" <<< "${GKE_CRED_JSON}"

in the `pull-request-schedule.yaml` of oyurs (IIRC).
Hm, not sure why that works, as you have `exportEnv: false` in the step before, retrieving the vault data.

Reply:
No, because if you removed the outer quotes the whole thing gets shell expansion.
`pull-request-schedule.yaml` has an associated `env:` block _for that step only_ which sets the variable.  In the quoted line, `"${GKE_CRED_JSON}"` is expanded by the shell, but only once; so the value (complete with quotes) is passed as-is as standard input.  As it is right now in this PR, `${{ … }}` is completed by the GitHub workflow (rather than the shell it spawns), so the shell is not aware of that and only sees the expanded values.

Reply:
@mook-as I now have a local commit going with your form (`cat ...`).

How can I test this ?

Could I
- make a fork of the repo, then
- merge the branch into the master of the fork,

and that would trigger GHA, but on the fork ?

Reply:
Yep, that should work; remember to apply the `pr-test-queue` label (or `pr-test-trigger` if you wan to skip the queuing step) to the PR to get it picked up.  Please ping me on Slack / 🚀 if you need assistance.

Reply:
Will do. Right now meeting...

Reply:
I missed the `has an associated env: block for that step only which sets` on my first look. Maybe gh page was not updated. That explains things and I have now put this into my spec too.